__tests__/unit/components/Inputs/TextInput.test.tsx

```diff
@@ -0,0 +1,119 @@
+import { fireEvent, render, screen } from '../../../testUtils/testing-utils'
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { TextInput } from 'components'
+
+describe('Text Area', () => {
+  describe('When type is text', () => {
+    const props = {
+      name: 'TextInput',
+      type: 'text',
+      required: true,
+      placeholder: 'Test Placeholder',
+      onChange: () => {},
+      onBlur: () => {}
+    }
+
+    it('should render the component', async () => {
+      render(<TextInput {...props} />)
+
+      const input = screen.getByTestId('TextInput')
+      expect(input).toBeInTheDocument()
+    })
+
+    it('should allow letters to be inputted', async () => {
+      render(<TextInput {...props} />)
+
+      const input = screen.getByTestId('TextInput') as HTMLInputElement
+
+      fireEvent.change(input, { target: { value: 'Great stuff' } })
+      expect(input.value).toBe('Great stuff')
+    })
+  })
+  describe('When type is number', () => {
+    const props = {
+      name: 'NumberInput',
+      type: 'number',
+      required: true,
+      placeholder: 'Test Placeholder',
+      onChange: () => {},
+      onBlur: () => {}
+    }
+
+    it('should render the component', async () => {
+      render(<TextInput {...props} />)
+
+      const input = screen.getByTestId('NumberInput')
+      expect(input).toBeInTheDocument()
+    })
+
+    it('should allow numbers to be inputted', async () => {
+      render(<TextInput {...props} />)
+
+      const input = screen.getByTestId('NumberInput') as HTMLInputElement
+      fireEvent.change(input, { target: { value: 23 } })
+      expect(input.value).toBe('23')
+    })
+
+    it('should not allow letters to be inputted', async () => {
+      render(<TextInput {...props} />)
+
+      const input = screen.getByTestId('NumberInput') as HTMLInputElement
+      fireEvent.change(input, { target: { value: 'Great stuff' } })
+      expect(input.value).toBe('')
+    })
+  })
+  describe('When type is checkbox', () => {
+    const props = {
+      name: 'CheckBox',
+      type: 'checkbox',
+      required: true,
+      placeholder: 'Test Placeholder',
+      onChange: () => {},
+      onBlur: () => {}
+    }
+
+    it('should render the component', async () => {
+      render(<TextInput {...props} />)
```

Comment:
is a dedicated CheckBox component now no?

Reply:
oh, yea I changed that in one of the last commits. Good catch!